src/libraries/Microsoft.Extensions.Hosting.WindowsServices/tests/UseWindowsServiceTests.cs

```diff
@@ -2,26 +2,159 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.IO;
+using System.Reflection;
+using System.ServiceProcess;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting.Internal;
+using Microsoft.Extensions.Hosting.WindowsServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.EventLog;
+using Microsoft.Extensions.Options;
 using Xunit;
 
 namespace Microsoft.Extensions.Hosting
 {
     public class UseWindowsServiceTests
     {
+        private static MethodInfo? _useWindowsServiceUncheckedMethod = null;
+
+        public static bool SupportsServiceBase
+        {
+            get
+            {
+                if (!PlatformDetection.IsWindows)
+                {
+                    return false;
+                }
+
+                try
+                {
+                    new ServiceBase();
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    // PlatformNotSupportedException : ServiceController enables manipulating and accessing Windows services and it is not applicable for other operating systems.
+                    //   at System.ServiceProcess.ServiceBase..ctor() in C:\dev\dotnet\runtime\artifacts\obj\System.ServiceProcess.ServiceController\Debug\net7.0\System.ServiceProcess.ServiceController.notsupported.cs:line 24
+                    //   at Microsoft.Extensions.Hosting.WindowsServices.WindowsServiceLifetime..ctor(IHostEnvironment environment, IHostApplicationLifetime applicationLifetime, ILoggerFactory loggerFactory, IOptions`1 optionsAccessor, IOptions`1 windowsServiceOptionsAccessor) in C:\dev\dotnet\runtime\src\libraries\Microsoft.Extensions.Hosting.WindowsServices\src\WindowsServiceLifetime.cs:line 26
+
+                    // REVIEW: This seems similar to https://github.com/dotnet/sdk/issues/16049 which was closed. I'm not sure why this is still happening in tests on Windows.
+                    // net462 tests can construct ServiceBase just fine, but not net7.0. Does anyone know what the real issue is here?
```

Comment:
@eerhardt Do you have any idea what's going on here? I see you commented on https://github.com/dotnet/sdk/issues/16049.

Reply:
My initial guess at the issue here is that System.ServiceProcess.ServiceController targets `net7.0-windows` and `net7.0`.

https://github.com/dotnet/runtime/blob/7654bf2c63d22dfdf197d54c1f4b7017aa556726/src/libraries/System.ServiceProcess.ServiceController/src/System.ServiceProcess.ServiceController.csproj#L3

The `windows`-specific version is what has the real code in it. The platform-agnostic version has PNSE all over, because you can't do windows things on platform-agnostic targets.

The tests only target `net7.0`. So when we build the tests, it picks the platform-agnostic version of `System.ServiceProcess.ServiceController`, which PNSEs.

So the fix here would be to target the tests for `windows` specifically. Check out here as an example:

https://github.com/dotnet/runtime/blob/7654bf2c63d22dfdf197d54c1f4b7017aa556726/src/libraries/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests.csproj#L2-L4

That's if you just want the tests to run on windows. If you want to run on all platforms, use:

https://github.com/dotnet/runtime/blob/7654bf2c63d22dfdf197d54c1f4b7017aa556726/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Microsoft.Extensions.Hosting.Unit.Tests.csproj#L3-L4